synapse/storage/databases/main/room.py

```diff
@@ -409,22 +409,32 @@ async def get_rooms_paginate(
             limit: maximum amount of rooms to retrieve
             order_by: the sort order of the returned list
             reverse_order: whether to reverse the room list
-            search_term: a string to filter room names by
+            search_term: a string to filter room names,
+                canonical alias and room ids by
         Returns:
             A list of room dicts and an integer representing the total number of
             rooms that exist given this query
         """
         # Filter room names by a string
         where_statement = ""
+        search_pattern = ""
         if search_term:
-            where_statement = "WHERE LOWER(state.name) LIKE ?"
+            where_statement = """
+                WHERE LOWER(state.name) LIKE ?
+                OR LOWER(state.canonical_alias) LIKE ?
+                OR LOWER(state.room_id) LIKE ?
```

Comment:
I don't think searching for substrings of room ids, or for case-insensitive matches of room ids, is a sensible thing to do. Room IDs aren't human-readable, and are meant to be opaque strings. It's dangerous to assume it even has a domain part - various spec-change proposals suggest changing that.

If people want to search by room id, they should enter the complete room ID and we should search for exactly that room id.

Reply:
Done